cpp/oneapi/dal/backend/primitives/rng/partial_shuffle.hpp

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+/*Partial Fisher-Yates Shuffle algorithm*/
+struct partial_shuffle {
+    void generate(ndview<size_t, 1>& result_array, size_t top);
+};
```

Comment:
Does it make sense to have structure here? Would the function be enough?

Reply:
Changed to function